### PR TITLE
`slack-vitess-r14.0.5`: remove legacy sharding, add MySQL 8.x backup test + compressors

### PR DESCRIPTION
**Pull request by timvaillancourt**
## Description

This PR backports the deprecation of legacy resharding functionality from v15

Backporting this early has 3 x benefits:
1. It removes our most flaky CI test: `Cluster (shardedrecovery_stress_verticalsplit_heavy)`
2. Backports for `vtorc` depend on these PRs to be merged
    - Doing this in separate PRs to limit changes
4. It removes unused code and CI tests
5. It adds missing tests for MySQL 8.x, which is being 🚢'd to prod
6. It adds support for different backup compressors, something we manually patch today

cc @ejortegau to validate that we are not using legacy resharding _(`vtworker`-based/no VReplication)_ functionality:
1. https://vitess.io/docs/archive/14.0/user-guides/historical/horizontal-sharding/ _(`SplitClone`)_
2. https://vitess.io/docs/archive/14.0/user-guides/historical/vertical-split/ _(`VerticalSplitClone`)_

If we are using legacy resharding, it's about to be removed in v15 😅 

## Related PR(s)

1. https://github.com/vitessio/vitess/pull/10278
2. https://github.com/vitessio/vitess/pull/10409
3. https://github.com/vitessio/vitess/pull/10459
4. https://github.com/vitessio/vitess/pull/10558
5. https://github.com/vitessio/vitess/pull/10691

## Checklist

-   [x] "Backport to:" labels have been added if this change should be back-ported
-   [x] Tests were added or are not required
-   [x] Did the new or modified tests pass consistently locally and on the CI
-   [x] Documentation was added or is not required

### CODE REVIEW

**Comment by timvaillancourt** [comment]
I've confirmed our `vtops` codebase never calls the legacy sharding functions. They exist in `vendor/` only:
```bash
tvaillancourt@tvailla-ltmxctu vtops % git grep SplitClone 
vendor/vitess.io/vitess/go/vt/binlog/binlogplayer/binlog_player.go:// workflow: documents the creator/manager of the stream. Example: 'SplitClone'.
vendor/vitess.io/vitess/go/vt/wrangler/keyspace.go:// SplitClone initiates a SplitClone workflow.
vendor/vitess.io/vitess/go/vt/wrangler/keyspace.go:func (wr *Wrangler) SplitClone(ctx context.Context, keyspace string, from, to []string) error {
vendor/vitess.io/vitess/go/vt/wrangler/keyspace.go:                     cmd := binlogplayer.CreateVReplicationState("VSplitClone", bls, "", binlogplayer.BlpStopped, primary.DbName())
vendor/vitess.io/vitess/go/vt/wrangler/keyspace.go:// VerticalSplitClone initiates a VerticalSplitClone workflow.
vendor/vitess.io/vitess/go/vt/wrangler/keyspace.go:func (wr *Wrangler) VerticalSplitClone(ctx context.Context, fromKeyspace, toKeyspace string, tables []string) error {
vendor/vitess.io/vitess/go/vt/wrangler/keyspace.go:     cmd := binlogplayer.CreateVReplicationState("VSplitClone", bls, "", binlogplayer.BlpStopped, primary.DbName())
vendor/vitess.io/vitess/go/vt/wrangler/keyspace.go:     return nil, nil, fmt.Errorf("neither Shard '%v' nor Shard '%v' have a 'SourceShards' entry. Did you successfully run vtworker SplitClone before? Or did you already migrate the MASTER type?", os.Left[0].ShardName(), os.Right[0].ShardName())
```

**Comment by timvaillancourt** [comment]
A note: the `salesforce-cla` bot is insane and thinks the author of the upstream PRs we backport needs to sign our CLA. Please ignore this CI

File: .github/workflows/cluster_endtoend_onlineddl_revert_mysql80.yml [hunk 1]
```diff
@@ -49,28 +38,28 @@ jobs:
...
-            - '.github/workflows/cluster_endtoend_vtgate_partial_keyspace.yml'
+            - '.github/workflows/**'
 
     - name: Set up Go
-      if: steps.skip-workflow.outputs.skip-workflow == 'false' && steps.changes.outputs.end_to_end == 'true'
-      uses: actions/setup-go@v3
+      if: steps.changes.outputs.end_to_end == 'true'
+      uses: actions/setup-go@v2
       with:
-        go-version: 1.19.3
```
Comment (timvaillancourt) [inline .github/workflows/cluster_endtoend_onlineddl_revert_mysql80.yml:58]: @maksimov these backports change `go-version` from `1.19.3` -> `1.18.7`. Were these changes you added and do we need to keep them? 🙇

**Review by rvrangel — APPROVED** [review]

File: .github/workflows/cluster_endtoend_onlineddl_revert_mysql80.yml (hunk 1 above)
Comment (ejortegau) [reply .github/workflows/cluster_endtoend_onlineddl_revert_mysql80.yml:58]: I believe we need to keep it at 1.19.3+ for security vulnerabilities, but it's better to double check with Stas once he's back.

**Review by ejortegau — APPROVED** [review]

**Pull request merged**